test/integration/testpaths/testpath_disable_enable_zone.py

```diff
@@ -0,0 +1,485 @@
+                -"The zone is not deletable because there are servers running in this zone"
+
+        """
+        # Step 1
+        vm_user = VirtualMachine.create(
+            self.userapiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype)
+
+        vm_root = VirtualMachine.create(
+            self.apiclient,
+            self.testdata["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id,
+            mode=self.zone.networktype)
+
+        cmd = updateZone.updateZoneCmd()
+        cmd.id = self.zone.id
+        cmd.allocationstate = "Disabled"
+        self.apiclient.updateZone(cmd)
+        zoneList = Zone.list(self.apiclient, id=self.zone.id)
+
+        self.assertEqual(zoneList[0].allocationstate,
+                         "Disabled",
+                         "Check if the zone is in disabled state"
+                         )
+        # Verify the exsisting vms should be running
+        self.assertEqual(vm_user.state,
+                         "Running",
+                         "Verify that the vm is running")
+
+        self.assertEqual(vm_root.state,
+                         "Running",
+                         "Verify that the vm is running")
+
+        time.sleep(30)
+        vm_root.stop(self.apiclient)
+        vm_user.stop(self.apiclient)
+
+        root_state = self.dbclient.execute(
+                "select state from vm_instance where name='%s'" %
+                vm_root.name)[0][0]
+
+        user_state = self.dbclient.execute(
+                "select state from vm_instance where name='%s'" %
+                vm_user.name)[0][0]
+
+        self.assertEqual(root_state,
+                "Stopped",
+                "verify that vm should stop")
+
+        self.assertEqual(user_state,
+                "Stopped",
+                "verify that vm should stop")
+
+        root_volume = list_volumes(
+            self.userapiclient,
+            virtualmachineid=vm_root.id,
+            type='ROOT',
+            listall=True
+        )
+
+        snap = Snapshot.create(
+            self.apiclient,
+            root_volume[0].id)
+
+        self.assertNotEqual(snap,
+                            None,
+                            "Verify that admin should snashot")
+
+        snapshots = list_snapshots(
+            self.apiclient,
+            volumeid=root_volume[0].id,
+            listall=True)
+
+        template_from_snapshot = Template.create_from_snapshot(
+            self.apiclient,
+            snapshots[0],
+            self.testdata["template_2"])
```

Comment:
use "privatetemplate" dict as it will create the template which is private to only that particular account. Otherwise, other test cases might use the same template and there might be a problem during cleanup operation.